### PR TITLE
feature contact/lead

### PR DESCRIPTION
**Pull request by dora-jurcevic**

### CODE REVIEW

File: components/global/ContactForm.vue [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      type: String,
+      default: ''
+    },
+  },
+  methods: {
+    save(e: Event) {
+      e.preventDefault()
+      this.submitted = true
+      let lead = this.lead
+      this.$emit('lead',lead)
```
Comment (thibaultrey) [inline components/global/ContactForm.vue:87]: Why this component use event to send the API Request ?

File: locales/en-US.json [hunk 2]
```diff
@@ -266,5 +266,25 @@
     "search": "Search",
     "cart": "Cart",
     "account": "Account"
+  },
```
Comment (thibaultrey) [inline locales/en-US.json:269]: is-it translated for the other translations files ?

File: pages/contact.vue [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    'contact-form': ContactForm
+  },
+  data(){
+    return {
+      thankyouMessage: false
+    }
+  },
+  methods: {
+    async sendLead(lead: Lead) {
+      const leadsService = useShopinvaderService('leads')
```
Comment (thibaultrey) [inline pages/contact.vue:37]: I'm not sure to understand why it's better to call API request here instead of directly in the component

**Review by thibaultrey — CHANGES_REQUESTED** [review]

File: components/global/ContactForm.vue [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          </div>
+        </slot>
+        <div>
+          <form v-if="lead" @submit="save">
+            <slot name="body">
+              <div class="form-fieldset">
+                <select v-model="lead.name" class="form-fieldset__select" required>
+                  <option  selected > {{$t('contact.subject')}} </option>
+                  <option :value="$t('contact.first_option')">{{$t('contact.first_option')}} </option>
+                  <option :vlaue="$t('contact.second_option')">{{$t('contact.second_option')}}</option>
```
[the remark this answers is not in the document]
Comment (dora-jurcevic) [reply components/global/ContactForm.vue]: fixed

File: components/global/ContactForm.vue [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                </select>
+                
+              </div>
+              <div class="form-fieldset ">
+                <input class="form-fieldset__input " type="text" :placeholder="$t('contact.contact_name')"  v-model="lead.contact_name" required>
```
[the remark this answers is not in the document]
Comment (dora-jurcevic) [reply components/global/ContactForm.vue]: refactored

File: components/global/ContactForm.vue [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+   dataPolicyPage :{
+      required: false,
+      type: String,
+      default: ''
+    },
+  },
+  methods: {
+    save(e: Event) {
+      e.preventDefault()
+      this.submitted = true
```
[the remark this answers is not in the document]
Comment (dora-jurcevic) [reply components/global/ContactForm.vue]: loader added on contact page

File: locales/en-US.json (hunk 2 above)
Comment (dora-jurcevic) [reply locales/en-US.json:269]: translations refactored, account.address translations used

File: pages/contact.vue (hunk 3 above)
Comment (dora-jurcevic) [reply pages/contact.vue:43]: not sure either, but we can imagine all the websites will have a contact page. I can also refactor it to be taken into account on the component.

**Review by thibaultrey — APPROVED** [review]

**Pull request merged**